static/skin/viewer.js

```diff
@@ -388,23 +392,57 @@ function getUserLanguage() {
       || DEFAULT_UI_LANGUAGE;
 }
 
+function makeURL(search, hash) {
+  let url = location.origin + location.pathname;
+  if (search != "") {
+    url += '?' + search;
+  }
+
+  url += (hash[0] == '#' ? hash : '#' + hash);
+  return url;
+}
+
 function initUILanguageSelector() {
   const languageSelector = document.getElementById("ui_language");
-  const userLang = getUserLanguage();
   for (const lang of uiLanguages ) {
     const lang_name = Object.getOwnPropertyNames(lang)[0];
     const lang_code = lang[lang_name];
-    const is_selected = lang_code == userLang;
+    const is_selected = lang_code == viewerState.uiLanguage;
     languageSelector.appendChild(new Option(lang_name, lang_code, is_selected, is_selected));
   }
 }
 
+function updateUILanguageSelector(userLang) {
+  console.log(`updateUILanguageSelector(${userLang})`);
+  const languageSelector = document.getElementById("ui_language");
+  for (const opt of languageSelector.children ) {
+    if ( opt.value == userLang ) {
+      opt.selected = true;
+    }
```

Comment:
Shouldn't we "unselect" the previous one ?

Reply:
The previous one is unselected automatically (since this is not in a multi-choice control).